### DISCUSSION TITLE
Custom Schematics Question

### DISCUSSION THREAD

**Discussion by Robw94** [cyrilletuzi/vscode-angular-schematics#935] [Q&A]
Hi,

How do I get the custom schematics to work? 

It generates all 5 files when I want to add options

{
  "id": "angular-standalone-component",
  "label": "Standalone Component",
  "collection": "Angular Components",
  "description": "Generate a standalone Angular component with inline template or separate HTML file, and CSS or SCSS",
  "options": {
    "externalTemplate": {
      "description": "Do you want an external template (HTML file) instead of an inline template?",
      "type": "boolean",
      "default": false
    },
    "style": {
      "description": "What type of style do you want?",
      "type": "string",
      "enum": ["css", "scss"],
      "enumDescriptions": ["Native CSS styles", "Sass styles"],
      "default": "scss"
    },
    "createTests": {
      "description": "Do you want to generate a test file?",
      "type": "boolean",
      "default": false
    }
  },
  "mainFile": "{{fileName}}.component.ts",
  "pathSubfolder": true
}

I cannot find anywhere in the documentation for this?

https://github.com/cyrilletuzi/vscode-angular-schematics/blob/main/walkthroughs/customSchematics.md

So if the option externalTemplate = false, then do not generate html..

The option is working so it either does templateUrl or template, salso the scss or cs, it is just generating all 5 files when I only want 3 or 4

In

**Comment by cyrilletuzi** [comment]
Hello @Robw94,

I sent you an email to the address you filled in your Gumroad subscription, to help you more efficiently and quickly.
